### DISCUSSION TITLE
Rotation around a fixed point

### DISCUSSION THREAD

**Discussion by amelillo** [senshu/Sozi#620] [Q&A]
Hi!
I was wondering about the way Sozi manages rotations.
Example:
I usually animate geometrical constructions, which often make use of a compass.
When I let Sozi manage the rotation of a svg compass to trace a small arc it behaves like a regular compass.
But if I have to show the compass drawing a whole circle or a big arc, the arc is shown but the compass doesn't rotate around its needle point; instead, it tumbles across the sheet to the final position as if the arc was drawn magically and the compass was just removed and put again in place.
I tried editing the compass in Inkscape and set the center of rotation onto the needle but it doesn't solve the problem.
By now, I'm just removing the compass from the presentation, letting the arcs appear by themselves, but it would be cool to show the compass.

Any hint?

![costruzioni02](https://user-images.githubusercontent.com/27341818/161396490-a226f6c5-46b3-430b-87fe-43ec44f62974.svg)

[costruzioni02.sozi.zip](https://github.com/sozi-projects/Sozi/files/8403174/costruzioni02.sozi.zip)

**Comment by aumouvantsillage** [comment]
> I was wondering about the way Sozi manages rotations.

During a typical transition, the camera follows a straight line from the center of the start frame to the center of the destination frame.
While moving, it rotates around the current point on this line.
I remember that I tried other strategies before settling on this one: I think that it felt more relevant for a presentation tool to prioritize the translation.

Maybe your use case could benefit from the "transition path" feature, where the camera is constrained to follow a given path during a transition. There is a demo where this feature is used as a workaround to rotate around a fixed point. Follow these links to [see the presentation](https://sozi-projects.github.io/Sozi-demos/transition-path/transition-path.sozi.html), and to [browse the source files](https://github.com/sozi-projects/Sozi-demos/tree/master/transition-path).

**Reply by DesmoricFR** [reply]
Bonjour,
Je suis très intéressé par la possibilité de faire une rotation autour d’un point fixe. J’ai examiné la présentation et les fichiers source, mais je n’ai pas compris sur quel chemin vous vous appuyez pour faire la rotation ni les paramétrages à effectuer. Pourriez-vous les expliquer ?

**Reply by aumouvantsillage** [reply]
Le chemin est celui en bleu clair dans le document SVG. Il a la forme d'un arc de cercle avec des flèches aux extrémités.
Les paramétrages à effectuer sont :
* Dans la section "Calque", champ "ID de l'élément à utiliser comme contour", indiquer l'identifiant  de chaque rectangle rouge : "rect-day" pour la vue 1, "rect-night" pour la vue 2.
* Dans section "Transition", champ "ID du chemin", indiquer l'identifiant : "path-day-to-night".

**Reply by DesmoricFR** [reply]
Bonjour,

Merci pour ces explications. Je vais adapter cette méthode à mon document : je souhaite faire « tourner » une roue à 360° sur son centre, en 4 étapes.
Je vous donnerai des nouvelles bientôt 😃.

**Reply by Moini** [reply]
Cool, I hadn't known about the transition path option! (and it already exists for so long...)
